### PR TITLE
feat: don't link parents if relationship to child is invalid

### LINKED ISSUE #10222: Add logic to skip linking invalid parent child relationships
- The `InOrderLinker` should check and not link events that have an invalid parent-child relationship
- The 2 checks currently relevant are:
    - time created
    - generation

### PR DESCRIPTION
**Pull request by alittley**
- closes #10222

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java [hunk 1]
```diff
@@ -51,6 +62,19 @@ public class InOrderLinker {
      */
     private static final int INITIAL_CAPACITY = 1024;
 
+    /**
+     * The minimum period between log messages for a specific mode of failure.
+     */
+    private static final Duration MINIMUM_LOG_PERIOD = Duration.ofMinutes(1);
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:68]: I'm seeing this in a lot of places.  Does this need to be configurable, a default value for rate limited logs?

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:68]: I'm asking the question for discussion purposes, not making a definite change request.

**Review by edward-swirldslabs — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java [hunk 2]
```diff
@@ -81,17 +105,117 @@ public class InOrderLinker {
     /**
      * Constructor
      *
+     * @param platformContext    the platform context
+     * @param time               a source of time
      * @param intakeEventCounter keeps track of the number of events in the intake pipeline from each peer
      */
-    public InOrderLinker(@NonNull final IntakeEventCounter intakeEventCounter) {
+    public InOrderLinker(
+            @NonNull final PlatformContext platformContext,
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:113]: If we just need the metrics, can we pass is just the metrics? I am not a big fan of passing the context everywhere, it makes the dependencies less obvious and complicates unit tests

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java (hunk 1 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:68]: @edward-swirldslabs I created an issue for this https://github.com/hashgraph/hedera-services/issues/10299

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java (hunk 2 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:113]: Based on Kelly's comment in slack, leaving this as the entire context

**Pull request merged**